pkg/operator/deploy/deploy_test.go

```diff
@@ -0,0 +1,118 @@
+package deploy
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"testing"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/cmp"
+)
+
+func TestIngressProfiles(t *testing.T) {
+
+	type test struct {
+		name    string
+		oc      func() *api.OpenShiftClusterProperties
+		want    string
+		wantErr string
+	}
+
+	for _, tt := range []*test{
+		{
+			name: "default IngressProfile",
+			oc: func() *api.OpenShiftClusterProperties {
+				return &api.OpenShiftClusterProperties{
+					IngressProfiles: []api.IngressProfile{
+						{
+							Name: "default",
+							IP:   "1.2.3.4",
+						},
+					},
+				}
+			},
+			want:    "1.2.3.4",
+			wantErr: ``,
+		},
+		{
+			name: "Multiple IngressProfiles, pick default",
+			oc: func() *api.OpenShiftClusterProperties {
+				return &api.OpenShiftClusterProperties{
+					IngressProfiles: []api.IngressProfile{
+						{
+							Name: "custom-ingress",
+							IP:   "1.1.1.1",
+						},
+						{
+							Name: "default",
+							IP:   "1.2.3.4",
+						},
+						{
+							Name: "not-default",
+							IP:   "1.1.2.2",
+						},
+					},
+				}
+			},
+			want:    "1.2.3.4",
+			wantErr: ``,
+		},
+		{
+			name: "Single Ingress Profile, No Default",
+			oc: func() *api.OpenShiftClusterProperties {
+				return &api.OpenShiftClusterProperties{
+					IngressProfiles: []api.IngressProfile{
+						{
+							Name: "custom-ingress",
+							IP:   "1.1.1.1",
+						},
+					},
+				}
+			},
+			want:    "1.1.1.1",
+			wantErr: ``,
+		},
+		{
+			name: "Multiple Ingress Profiles, No Default",
+			oc: func() *api.OpenShiftClusterProperties {
+				return &api.OpenShiftClusterProperties{
+					IngressProfiles: []api.IngressProfile{
+						{
+							Name: "custom-ingress",
+							IP:   "1.1.1.1",
+						},
+						{
+							Name: "not-default",
+							IP:   "1.1.2.2",
+						},
+					},
+				}
+			},
+			want:    "1.1.1.1",
+			wantErr: ``,
+		},
+		{
+			name: "No IngressProfiles, Error",
+			oc: func() *api.OpenShiftClusterProperties {
+				return &api.OpenShiftClusterProperties{
+					IngressProfiles: []api.IngressProfile{},
```

Comment:
nit: should be super easy to add the final case of `nil` `IngressProfiles` :)

Reply:
fixed, thank you